.github/workflows/publish.yml

```diff
@@ -0,0 +1,57 @@
+name: Build & Publish
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    name: release
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Use Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: 14
+    # Run install dependencies
+    - name: Install dependencies
+      run: npm install
+    # Run tests
+    - name: Build 
+      run: npm run compile
+    - name: Get current package version
+      id: package_version
+      uses: martinbeentjes/npm-get-version-action@v1.1.0
+    - name: Check version is mentioned in Changelog
+      id: changelog_reader
+      uses: mindsers/changelog-reader-action@v2
+      with:
+        validation_depth: 10
+        version: ${{ steps.package_version.outputs.current-version }}
+        path: 'CHANGELOG.md'
+    - name: Create a Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+      with:
+        tag_name : ${{ steps.package_version.outputs.current-version}}
+        release_name: ${{ steps.package_version.outputs.current-version}}
+        body: Publish ${{ steps.changelog_reader.outputs.changes }}
+    - name: Create vsix and publish to marketplace
+      id: create_vsix
+      uses: HaaLeo/publish-vscode-extension@v0
```

Comment:
I don't know if this is actually a potential problem or not, but at this point would this HaaLeo extension have access to the VS_MARKETPLACE_TOKEN? Are there any risks in using an external action here?

Reply:
💡☕️ Thank for mention and its good to archive this for any future readers, this is just a GH action for vscode publish which uses a GH Secret Variable which cannot be logged.

reference: https://github.com/marketplace/actions/publish-vs-code-extension and for GH secret variable details here: https://docs.github.com/en/actions/security-guides/encrypted-secrets